frontend/src/components/SortServices.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react'
+import { Badge, IconButton, Menu, MenuItem } from '@material-ui/core'
+import { Icon } from './Icon'
+import { useDispatch } from 'react-redux'
+import { Dispatch } from '../store'
+import { makeStyles } from '@material-ui/core/styles'
+import { fontSize } from '@remote.it/components/lib/styles/variables'
+
+const options = [
+  { value: 0, name: 'Alphabetical(A-Z)' },
+  { value: 1, name: 'Alphabetical(Z-A)' },
+  { value: 2, name: 'Creation Date (Newest)' },
+  { value: 3, name: 'Creation Date (Oldest)' },
+]
+
+export const SortServices: React.FC = () => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const { devices } = useDispatch<Dispatch>()
+  const open = Boolean(anchorEl)
+  const css = useStyles()
+  const [icon, setIcon] = React.useState('sort-amount-down')
+  const [dot, setDot] = React.useState(false)
+  const [optionSelected, setOptionSelected] = React.useState(0)
+
+  useEffect(() => {
+    servicesSort(0)
+  }, [])
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+
+  const servicesSort = async (type: number) => {
+    setOptionSelected(type)
+    let sortService
+    switch (type) {
+      case 0:
+        sortService = (a: IService, b: IService) =>
+          a.name.toLowerCase() > b.name.toLowerCase() ? 1 : a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 0
+        break
+      case 1:
+        sortService = (a: IService, b: IService) =>
+          a.name.toLowerCase() < b.name.toLowerCase() ? 1 : a.name.toLowerCase() > b.name.toLowerCase() ? -1 : 0
+        break
+      case 2:
+        sortService = (a: IService, b: IService) => (a.createdAt > b.createdAt ? 1 : -1)
+        break
+      case 3:
+        sortService = (a: IService, b: IService) => (a.createdAt < b.createdAt ? 1 : -1)
+        break
+    }
```

Comment:
What about adding these sort functions to the options on line 9 above, so that you can then save the option index to the device model. That way you can read the sort function AND the selected option out of the model at the same time so that you can restore the icon state when leaving and returning to the page.